gcloud-java-compute/src/main/java/com/google/gcloud/compute/RawImageConfiguration.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.compute.model.Image;
+import com.google.common.base.MoreObjects;
+
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine raw image configuration. This class can be used to create images from a
+ * Google Cloud Storage URL where the disk image is stored.
+ */
+public class RawImageConfiguration extends ImageConfiguration {
+
+  private static final long serialVersionUID = 8160447986545005880L;
+
+  private final ContainerType containerType;
+  private final String sha1;
+  private final String source;
+  private final Long archiveSizeBytes;
+
+  /**
+   * The format used to encode and transmit the block device. The only supported value is
+   * {@code TAR}. This is just a container and transmission format, not a runtime format.
+   */
+  public enum ContainerType {
+    TAR
+  }
+
+  /**
+   * A builder for {@code RawImageConfiguration} objects.
+   */
+  public static final class Builder
+      extends ImageConfiguration.Builder<RawImageConfiguration, Builder> {
+
+    private ContainerType containerType;
+    private String sha1;
+    private String source;
+    private Long archiveSizeBytes;
+
+    private Builder() {}
+
+    private Builder(RawImageConfiguration imageConfiguration) {
+      super(imageConfiguration);
+      this.containerType = imageConfiguration.containerType;
+      this.sha1 = imageConfiguration.sha1;
+      this.source = imageConfiguration.source;
+      this.archiveSizeBytes = imageConfiguration.archiveSizeBytes;
+    }
+
+    private Builder(Image imagePb) {
+      super(imagePb);
+      if (imagePb.getRawDisk().getContainerType() != null) {
+        this.containerType = ContainerType.valueOf(imagePb.getRawDisk().getContainerType());
+      }
+      this.sha1 = imagePb.getRawDisk().getSha1Checksum();
+      this.source = imagePb.getRawDisk().getSource();
+      this.archiveSizeBytes = imagePb.getArchiveSizeBytes();
```

Comment:
I checked and in fact this field is set also for image created from an existing disk (i.e. with DiskImageConfiguration). Weird that it's documented as: 

> Size of the image tar.gz archive stored in Google Cloud Storage (in bytes).

That made me think that it only applied when the disk was created from a source in Storage.